src/main/kotlin/io/snyk/plugin/ui/jcef/OpenFileLoadHandlerGenerator.kt

```diff
@@ -35,32 +35,76 @@ class OpenFileLoadHandlerGenerator(snykCodeFile: SnykCodeFile) {
         return JBCefJSQuery.Response("success")
     }
 
+    fun isDarcula(): Boolean {
+        val scheme = EditorColorsManager.getInstance().globalScheme
+        return scheme.name.contains("Darcula")
+    }
+
     fun generate(jbCefBrowser: JBCefBrowserBase): CefLoadHandlerAdapter {
         val openFileQuery = JBCefJSQuery.create(jbCefBrowser)
         openFileQuery.addHandler { openFile(it) }
 
         return object : CefLoadHandlerAdapter() {
             override fun onLoadEnd(browser: CefBrowser, frame: CefFrame, httpStatusCode: Int) {
                 if (frame.isMain) {
-                    browser.executeJavaScript(
-                        "window.openFileQuery = function(value) {" +
-                            openFileQuery.inject("value") +
-                            "};",
-                        browser.url, 0
-                    );
+                    val script = """
+                    (function() {
+                        if (window.openFileQuery) {
+                            return;
+                        }
+                        window.openFileQuery = function(value) { ${openFileQuery.inject("value")} };
+                        // Attach a single event listener to the document
+                        document.addEventListener('click', function(e) {
+                            // Find the nearest ancestor
+                            var target = e.target.closest('.data-flow-clickable-row');
+                            if (target) {
+                                e.preventDefault();
+                                window.openFileQuery(target.getAttribute("file-path") + ":" +
+                                                     target.getAttribute("start-line") + ":" +
+                                                     target.getAttribute("end-line") + ":" +
+                                                     target.getAttribute("start-character") + ":" +
+                                                     target.getAttribute("end-character"));
+                            }
+                        });
+                    })();
+                """
```

Comment:
- Use event delegation for the `.data-flow-clickable-row` elements. This change reduces the
  number of event listeners attached to the DOM by binding a single listener on
  the document rather than individual listeners on each element.
- Prevent redefinition of the `openFileQuery` function initialising just once per page load.